### PR TITLE
refactor(fo): upgrade FO parser to use event classes

### PR DESCRIPTION
**Pull request by amv213**
## Issue

https://github.com/electricitymaps/electricitymaps-contrib/issues/6011

## Description

This PR upgrades the `FO` parser to use event classes, and adds tests to catch regressions.

### Double check

- [x] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [x] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

File: parsers/FO.py [hunk 1]
```diff
@@ -22,97 +22,72 @@
...
                 )
             # Power (MW)
             value = float(value.replace(",", "."))
-            data["production"][generation_type] = (
-                data["production"].get(generation_type, 0) + value
-            )
+            production_mix.add_value(generation_type, value)
         else:
-            # print 'Unhandled key %s' % key
             pass
```
Comment (VIKTORVAV99) [inline parsers/FO.py:78]: Pretty sure we can just delete these two lines. They do nothing.

File: parsers/FO.py [hunk 2]
```diff
@@ -22,97 +22,72 @@
...
 
-    data = validate(
-        data,
-        logger,
-        floor=ZONE_MAP[zone_key]["validation"]["floor"],
+    production_breakdown_list.append(
+        zoneKey=zone_key,
+        datetime=datetime.fromisoformat(obj["tiden"])
+        .replace(tzinfo=TIMEZONE)
+        .astimezone(timezone.utc),
```
Comment (VIKTORVAV99) [inline parsers/FO.py:84]: As timezone is not needed FYI. It will be normalised in the backend anyway.

**Review by VIKTORVAV99 — COMMENTED** [review]
Just some small notes. But looks good!

File: parsers/FO.py (hunk 1 above)
Comment (amv213) [reply parsers/FO.py:78]: In addition to production values aggregated by production mode that we handle above there are some extra keys which make their way through here (I think they are the itemised production values by power plant):

```
tiden
SdDiesel_E
FoVand_E
MyVand_E
HeVand_E
EiVand_E
StDiesel_E
StVand_E
NoDiesel_E
SorDiesel_E
HhVind_E
NeVind_E
VgDiesel_E
BoVand_E
SuSol_E
PoVind_E
RoVind_E
ViRoVind_E
SuStVind_E
NoStVind_E
SuVind_E
Minesto_E
Forka_E
VandH_E
OlieH_E
VindH_E
BiogasH_E
TidalH_E
VandSev_E
OlieSev_E
VindSev_E
SolSev_E
BiogasSev_E
TidalSev_E
SdDiesel_P
FoVand_P
MyVand_P
HeVand_P
EiVand_P
StDiesel_P
StVand_P
NoDiesel_P
SorDiesel_P
HhVind_P
NeVind_P
RoVind_P
ViRoVind_P
SuStVind_P
NoStVind_P
SuVind_P
VgDiesel_P
BoVand_P
SuSol_P
PoVind_P
Minesto_P
Forka_P
VandH_P
VindH_P
OlieH_P
BiogasH_P
TidalH_P
VandS_P
OlieS_P
VindS_P
SolS_P
VandSev_P
OlieSev_P
VindSev_P
SolSev_P
BiogasSev_P
TidalSev_P
Gront_P
Vatnid
Lomundaroyri
Heygadalur
Eidisvatn
Strandadalur
Ryskisvatn
Vatnsnesvatn
```

Comment (VIKTORVAV99) [reply parsers/FO.py:78]: Ah okey but then it should probably be continue instead of pass IMO as pass is usually a placeholder.

**Comment by VIKTORVAV99** [comment]
Will do a final review in just a bit.

**Pull request merged**